modules/store/src/action_creator.ts

```diff
@@ -0,0 +1,47 @@
+// ==========
+// = Adapted from: rex-tils
+// = https://github.com/Hotell/rex-tils
+// ==========
+
+/**
+ * ActionWithProps can exist in only 2 shapes; type only or type with properties
+ */
+type ActionWithProps<T extends string = string, P = void> = P extends void

```

Comment:
What about using this type as our "NgRx Action", in favor of https://github.com/ngrx/platform/blob/master/modules/store/src/models.ts#L1.